underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -206,19 +208,34 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri,
       clientConf.setSignerOverride(conf.get(PropertyKey.UNDERFS_S3_SIGNER_ALGORITHM));
     }
 
-    AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder
-        .standard()
+    AmazonS3ClientBuilder clientBuilder = AmazonS3Client.builder()
         .withCredentials(credentials)
         .withClientConfiguration(clientConf);
 
     if (conf.getBoolean(PropertyKey.UNDERFS_S3_DISABLE_DNS_BUCKETS)) {
       clientBuilder.withPathStyleAccessEnabled(true);
     }
 
+    boolean enableGlobalBucketAccess = true;
     AwsClientBuilder.EndpointConfiguration endpointConfiguration
         = createEndpointConfiguration(conf, clientConf);
     if (endpointConfiguration != null) {
       clientBuilder.withEndpointConfiguration(endpointConfiguration);
+      enableGlobalBucketAccess = false;
+    } else if (conf.isSet(PropertyKey.UNDERFS_S3_REGION)) {
+      try {
+        clientBuilder.withRegion(conf.get(PropertyKey.UNDERFS_S3_REGION));
+        enableGlobalBucketAccess = false;
+      } catch (SdkClientException e) {
+        LOG.error("S3 region {} cannot be recognized, "
+            + "fall back to use global bucket access with an extra HEAD request",
+            conf.get(PropertyKey.UNDERFS_S3_REGION), e);
+      }
+    }
+    if (enableGlobalBucketAccess) {
```

Comment:
I have always wondered why there is an extra HEAD request!!!

Reply:
! you mean our previous implementation always have an extra HEAD request???
our previous implementation doesn't set region at all...... i guess global bucket access is always enabled